### ISSUE TITLE
Goto deletes entire project if trying to save a magic word with æøå in it

### ISSUE THREAD

**Issue by technocake** [technocake/goto#41]
A little critical. 

There is unicode issues in both loading and saving the json if magic words contain non-ascii characters. 

A very very small sideeffect of this is that all project data is lost. 

Until this is patched - make sure to do regular backups of your ~/.goto folder

**Comment by Arxcis** [comment]
Bump! :+1: 

I did not have a regular backup of my ~/.goto folder. This just happened to me. All my precious shortcuts at work lost when trying to do `goto add testnøkkel <uri>` :(

**Comment by technocake** [comment]
I am sorry :(.  I believe I tried to send you this gist at a point in time: https://gist.github.com/technocake/0ae3baea61bec186e4eb44b971209238

**Comment by Arxcis** [comment]
@technocake Don't worry about it ;) Needless to say, I have never been a bigger fan of the `utf-8` fix than now

**Comment by technocake** [comment]
https://github.com/technocake/goto/pull/107 fixes this issue for all terminals that are using utf-8 as input encoding. (utf-8 is ascii compliant if someone have ascii terminals).

Other encodings will currently not work well with goto if special chars are used. 
I am working on a fix for this also - by using chardet to detect the input encoding. 

The same PR is also refactoring the internal data structure in goto - making it possible to add a git repo for each project, and make goto create commits for each change to the project files. 

The git part is soon to be implemented fully - a proof of concept is already made and working wonders.

**Comment by technocake** [comment]
The more I am reading about on the issue of detecting the right encodings in all environment, the more I want that to be outside the scope of goto and use a framework which excels in this already. 

`click.utils.echo` for output!

https://github.com/pallets/click/blob/master/click/utils.py#L167

**Comment by Arxcis** [comment]
Yes, lets just import that single function - `click.utils.echo` ❤️ We don't have to care about the rest of the framework.

**Comment by technocake** [comment]
we have to care about the input also, hoping we can use something clicky

**Comment by Arxcis** [comment]
If we are `python3`-only in 6 months? Do we still need something `clicky` then?

**Comment by technocake** [comment]
Similar bug has been detected. 
This takes top priority to fix.  Working on it.

Example:

```
goto add spørreundersøkelse "https://link/to/somewhere"
Traceback (most recent call last):
  File "/usr/local/bin/the_real_goto.py", line 11, in <module>
    load_entry_point('magicgoto', 'console_scripts', 'the_real_goto.py')()
  File "/Users/technocake/code/goto/goto/the_real_goto.py", line 40, in main
    print_utf8(err.message)
  File "/Users/technocake/code/goto/goto/gotomagic/text/text.py", line 46, in message
    "Missing warning message with name: {}".format(self.name))
Exception: Missing warning message with name: magic_could_not_be_saved
```

Environment:
goto version: 1.6.0
os: os-x Catalina 10.15.7
SHELL=/bin/zsh
TERM=xterm-256color
LANG=<not set>
LOCALE=<not set>
TERM_PROGRAM=Apple_Terminal

<img width="331" alt="image" src="https://user-images.githubusercontent.com/1152171/123234698-732a2400-d4db-11eb-99ee-c8f963275e53.png">
